flake.nix

```diff
@@ -155,6 +155,9 @@
               {
                 default = config.legacyPackages.llamaPackages.llama-cpp;
                 vulkan = config.packages.default.override { useVulkan = true; };
+                windows = config.legacyPackages.llamaPackages.llama-cpp.override {
+                  stdenv = pkgs.pkgsCross.mingwW64.stdenv;
+                };
```

Comment:
Oh wait! Actually, I'm surprised this worked. I never touched windows, but shouldn't we be taking the other dependencies from `pkgsCross` as well?

Reply:
Which would look something like `(pkgsCross.mingwW64.callPackage .devops/nix/scope.nix { }).llama-cpp`

Reply:
From my experimentation the only parts that are needed to cross compile for Windows is the MinGW executable and the libraries it comes with which is all in stdenv. I did consider using `pkgsCross.callPackage` instead of overwriting `stdenv` but wouldn't that mean that we cannot compile for windows _and_ CUDA/ROCM since we use `pkgsCuda.callPackage` and `pkgsRocm.callPackage` respectively?

Reply:
> but wouldn't that mean that we cannot compile for windows and CUDA/ROCM since we use pkgsCuda.callPackage and pkgsRocm.callPackage respectively?

Being able to construct a configuration with cuda+rocm and exposing a flake output for that configuration are two different things. The flake outputs are there mostly for show. The actual "useful" program is the `package.nix` (and the `scope.nix`) which are functions that you can call from arbitrary instances of Nixpkgs.

Those can be consumed e.g. as overlays

Reply:
But also linking the build platform's (linux's) CUDA libraries to a mingw executable would obviously fail

Reply:
```
inputs.nixpkgs = ...
inputs.llama-cpp = ...
outputs = { nixpkgs, llama-cpp, ... }: {
  packages.x86_64-linux.llamaWindowsCuda = (import nixpkgs { config.cudaSupport = true; overlays = [ llama-cpp.overlays.default ]; }).pkgsCross.llama-cpp;
}
```

et cetera (almost certainly broken, we haven't looked into the windows instances of cudaPackages yet)

Reply:
Ah, ok. So using `pkgsCross.mingwW64.callPackage` would still work? I'll look into doing that then.

Reply:
You can extend `legacyPackages` the same way it's done for cuda/rocm

Reply:
Yes I've tried doing this while using `pkgs.pkgsCross.mingwW64.callPackage` but I'm getting problems with packages that complain about the platform now. This will take some time to get working in the right way but I'm on it.

Reply:
(I'm also surprised we didn't have to use `makeScopeWithSplicing`)

Reply:
Alright after quite a bit of testing it seems that the best solution is to override `stdenv` - I've tested the binaries on a Windows system and they all seems to run correctly 👍

I think this should be ready to merge. Let me know if you think anything is missing.

Reply:
Hi, check out the patches I just pushed. The only thing that was broken in the cross-compilation set up was python

Reply:
You did it! I tried to use `pkgs.pkgsCross.mingwW64.callPackage` but I kept getting that python error - which you've fixed now. I can even compile for Vulkan and it works on our Windows machine.

<img width="787" alt="screenshot" src="https://github.com/ggerganov/llama.cpp/assets/6594598/c8504e3f-fe78-4280-b5f2-ecc749a14acc">